test/assignment4/Assignment4Test.java

```diff
@@ -0,0 +1,77 @@
+package assignment4;
+
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import javax.imageio.ImageIO;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+public class Assignment4Test {
+    
+    private static final String IMAGE_DIR = "pics";
+    private static final String[] IMAGE_LIST =
+        {"crosses.bmp", "diagonal.bmp", "disks.bmp", "letters.bmp", "shapes.bmp", "text_binary.bmp", "empty.bmp"};
+    private static final int[] NO_OF_ISLANDS = {9, 88, 22, 25, 145, 87, 0};
+    private static final List<Map> MAPS = new ArrayList<>();
+    
+    private static class Map {
+        
+        private final int height;
+        private final int width;
+        private final boolean[][] map;
+        
+        public Map(String filename) throws IOException {
+            BufferedImage img = ImageIO.read(new File(filename));
+            this.height = img.getHeight();
+            this.width = img.getWidth();
+            this.map = new boolean[height][width];
+            for (int i = 0; i < height; i++) {
+                for (int j = 0; j < width; j++) {
+                    this.map[i][j] = ((char) img.getRGB(j, i) == 0);
+                }
+            }
+        }
+        
+        public int getHeight() {
+            return this.height;
+        }
+        
+        public int getWidth() {
+            return this.width;
+        }
+        
+        public boolean[][] getMap() {
+            return this.map;
+        }
+    }
+    
+    @BeforeClass
+    public static void setUpClass() {

```

Comment:
No need to load images in `@BeforeClass` and store them in map. You could safely do it one by one from a helper method called by tests.

Always aim to have code as simple as possible, unless you have a very good reason to make it more complex :)